docs/benchmarkdotnet.md

```diff
@@ -0,0 +1,332 @@
+
+### Running In Process
+
+Sometimes the easiest way to run the benchmarks against the local build of any non-AOT .NET Runtime is to copy all the files into one place, build the app with `csc` and just run the executable.
+
+If you want to run the benchmarks in the same process, without creating a dedicated executable and process-level isolation, please use `--inProcess` (or just `-i`).
+
+Please use this option only when you are sure that the benchmarks you want to run **have no side effects**. Allocating managed memory has side effects!
+
+### CoreRun
+
+It's possible to benchmark private builds of CoreCLR/FX using CoreRun.
+
+```cmd
+dotnet run -f netcoreapp3.0 --coreRun $thePath
+```
+
+**Note:** You can provide more than 1 path to CoreRun. In such case, the first path will be the baseline and all the benchmarks are going to be executed for all CoreRuns you have specified.
+
+**Note:** If `CoreRunToolchain` detects that you have some older version of dependencies required to run the benchmarks in CoreRun folder, it's going to overwrite them with newer versions from the published app. It's going to do that in a shadow copy of the folder with CoreRun, so your configuration remains untouched.
+
+If you are not sure which assemblies gets loaded and used you can use the following code to find out:
+
+```cs
+[GlobalSetup]
+public void PrintInfo()
+{
+    var coreFxAssemblyInfo = FileVersionInfo.GetVersionInfo(typeof(Regex).GetTypeInfo().Assembly.Location);
+    var coreClrAssemblyInfo = FileVersionInfo.GetVersionInfo(typeof(object).GetTypeInfo().Assembly.Location);
+
+    Console.WriteLine($"// CoreFx version: {coreFxAssemblyInfo.FileVersion}, location {typeof(Regex).GetTypeInfo().Assembly.Location}, product version {coreFxAssemblyInfo.ProductVersion}");
+    Console.WriteLine($"// CoreClr version {coreClrAssemblyInfo.FileVersion}, location {typeof(object).GetTypeInfo().Assembly.Location}, product version {coreClrAssemblyInfo.ProductVersion}");
+}
+```
+
+### dotnet cli
+
+You can also use any dotnet cli to build and run the benchmarks. 
+
+```cmd
+dotnet run -f netcoreapp3.0 --cli "C:\Projects\performance\.dotnet\dotnet.exe"
+```
+
+This is very useful when you want to compare different builds of .NET Core SDK.
+
+### Private CLR Build
+
+It's possible to benchmark a private build of .NET Runtime. You just need to pass the value of `COMPLUS_Version` to BenchmarkDotNet. You can do that by either using `--clrVersion $theVersion` as an argument or `Job.ShortRun.With(new ClrRuntime(version: "$theVersiong"))` in the code.
+
+So if you made a change in CLR and want to measure the difference, you can run the benchmarks with:
+
+```cmd
+dotnet run -f net472 -- --clrVersion $theVersion
+```
+
+More info can be found [here](https://github.com/dotnet/BenchmarkDotNet/issues/706).
+
+### Private CoreRT Build
+
+To run benchmarks with private CoreRT build you need to provide the `IlcPath`. Example:
+
+```
```

Comment:
`MD040/fenced-code-language: Fenced code blocks should have a language specifiedmarkdownlint(MD040)`

Reply:
It should be: ```cmd